Source/Python/pyWarpX.cpp

```diff
@@ -113,6 +115,10 @@ PYBIND11_MODULE(PYWARPX_MODULE_NAME, m) {
     m.def("check_gridding_for_RZ_spectral", &CheckGriddingForRZSpectral,
         "Ensure that the grid is setup appropriately with using the RZ spectral solver");
 
+    // Expose the WarpX instance
+    m.def("get_instance", &WarpX::GetInstance,
+        "Return a reference to the WarpX object.");
```

Comment:
Since the WarpX class is a singleton, we might not need `py::return_value_policy::reference` here.
But looking at our default usage in C++, `auto & warpx = WarpX::GetInstance();`, we better add it:

https://pybind11.readthedocs.io/en/stable/advanced/functions.html#return-value-policies
```suggestion
        py::return_value_policy::reference,
        "Return a reference to the WarpX object.");
```